### ISSUE TITLE
Openjdk tests failed with msg 'Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled'

### ISSUE THREAD

**Issue by sophia-guo** [AdoptOpenJDK/openjdk-build#893]
Recently ( from Jan 2, 2019) random subset of of following tests intermittently fails on HS with unexpected errors. The failure is:

```
21:06:26 #
21:06:26 # A fatal error has been detected by the Java Runtime Environment:
21:06:26 #
21:06:26 #  SIGSEGV (0xb) at pc=0x00007fb8245387ca, pid=28972, tid=29819
21:06:26 #
21:06:26 # JRE version: OpenJDK Runtime Environment (11.0.2+9) (build 11.0.2+9-201902021908)
21:06:26 # Java VM: OpenJDK 64-Bit Server VM (11.0.2+9-201902021908, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
21:06:26 # Problematic frame:
21:06:26 # J 3174 c2 java.security.MessageDigest$Delegate.engineUpdate([BII)V java.base@11.0.2 (11 bytes) @ 0x00007fb8245387ca [0x00007fb8245382e0+0x00000000000004ea]
21:06:26 #
21:06:26 # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
21:06:26 #
21:06:26 # An error report file with more information is saved as:
21:06:26 # /home/jenkins/workspace/openjdk11_hs_openjdktest_x86-64_linux/jvmtest/openjdk_regression/work/scratch/2/hs_err_pid28972.log
21:06:26 Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
21:06:26 #
21:06:26 # If you would like to submit a bug report, please visit:
21:06:26 #   https://github.com/AdoptOpenJDK/openjdk-build/issues
21:06:26 #
21:06:26 STDERR:

```

https://ci.adoptopenjdk.net/view/Test_openjdk/job/openjdk11_hs_openjdktest_x86-64_linux/153/consoleFull
https://ci.adoptopenjdk.net/view/Test_openjdk/job/openjdk11_hs_openjdktest_x86-64_linux/154/consoleFull
https://ci.adoptopenjdk.net/view/Test_openjdk/job/openjdk11_hs_openjdktest_x86-64_linux/156/consoleFull

java/net/httpclient/AsFileDownloadTest.java	
java/net/httpclient/InvalidInputStreamSubscriptionRequest.java	
java/net/httpclient/StreamingBody.java
java/net/httpclient/http2/ContinuationFrameTest.java
java/net/httpclient/SpecialHeadersTest.java
java/net/httpclient/http2/BadHeadersTest.java
java/net/httpclient/http2/BadHeadersTest.java
java/net/httpclient/DigestEchoClientSSL.java
java/net/httpclient/ResponseBodyBeforeError.java

It only happened to HS( by now I didn't see the issue with Openj9). According to the msg it could be a build issue?

**Comment by sophia-guo** [comment]
Another one:
java/net/httpclient/ResponsePublisher.java

https://ci.adoptopenjdk.net/view/Test_openjdk/job/openjdk11_hs_openjdktest_x86-64_linux/182/console

**Comment by makarid** [comment]
Hello, i am also getting the same crash on my application right after i migrate it to java 11 from java 8. Now i am using java 11.0.2. Does anyone knows what it is and why it is happening? The msg has always different class in which the application has crash.

**Comment by tsarenkotxt** [comment]
Hello, I am also getting the similar crash `Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled`:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f18285f1ca0, pid=6345, tid=6376
#
# JRE version: OpenJDK Runtime Environment (11.0.6+10) (build 11.0.6+10)
# Java VM: OpenJDK 64-Bit Server VM (11.0.6+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# J 25429 c2 org.springframework.boot.context.properties.source.ConfigurationPropertyName.defaultElementEquals(Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;Lorg/springframework/boot/context/properties/source/ConfigurationPropertyName$Elements;I)Z (436 bytes) @ 0x00007f18285f1ca0 [0x00007f18285f1c40+0x0000000000000060]
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P" (or dumping to /tmp/jenkins-ecb52fa2/workspace/master/core.6345)
#
# An error report file with more information is saved as:
# /tmp/jenkins-ecb52fa2/workspace/master/hs_err_pid6345.log
Compiled method (c2)  129517 25429       4       org.springframework.boot.context.properties.source.ConfigurationPropertyName::defaultElementEquals (436 bytes)
 total in heap  [0x00007f18285f1890,0x00007f18285f8838] = 28584
 relocation     [0x00007f18285f1a08,0x00007f18285f1c28] = 544
 main code      [0x00007f18285f1c40,0x00007f18285f6520] = 18656
 stub code      [0x00007f18285f6520,0x00007f18285f6558] = 56
 oops           [0x00007f18285f6558,0x00007f18285f6568] = 16
 metadata       [0x00007f18285f6568,0x00007f18285f6630] = 200
 scopes data    [0x00007f18285f6630,0x00007f18285f7ad8] = 5288
 scopes pcs     [0x00007f18285f7ad8,0x00007f18285f8578] = 2720
 dependencies   [0x00007f18285f8578,0x00007f18285f8588] = 16
 handler table  [0x00007f18285f8588,0x00007f18285f85b8] = 48
 nul chk table  [0x00007f18285f85b8,0x00007f18285f8838] = 640
Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
#
# If you would like to submit a bug report, please visit:
#   https://github.com/AdoptOpenJDK/openjdk-support/issues
#
```
Should I provide a full error report?

**Comment by smlambert** [comment]
Based on the number of similar cases reported, adding this to the top priorities list.

**Comment by karianna** [comment]
@tsarenkotxt - Is this crashing consistently?  If so can you try our nightly 11.0.7 build and/or the upstream build (you can find that on our main page)?

**Comment by StephanWald** [comment]
I can confirm that the upstream build fixes at least my problem (pristine Windows server):

```
C:\Program Files\AdoptOpenJDK>openjdk-11.0.7_9\bin\java -version
openjdk version "11.0.7-ea" 2020-04-14
OpenJDK Runtime Environment 18.9 (build 11.0.7-ea+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.7-ea+9, mixed mode)

C:\Program Files\AdoptOpenJDK>jdk-11.0.6.10-hotspot\bin\java -version
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ILLEGAL_INSTRUCTION (0xc000001d) at pc=0x000001d2c1e20c06, pid=4288, tid=10044
#
# JRE version:  (11.0.6+10) (build )
# Java VM: OpenJDK 64-Bit Server VM (11.0.6+10, mixed mode, tiered, compressed oops, g1 gc, windows-amd64)
# Problematic frame:
# v  ~StubRoutines::call_stub
#
# No core dump will be written. Failed to create minidump file (0x5).
#
# An error report file with more information is saved as:
# C:\Users\STEPH\AppData\Local\Temp\2\\hs_err_pid4288.log
Could not load hsdis-amd64.dll; library not loadable; PrintAssembly is disabled
#
# If you would like to submit a bug report, please visit:
#   https://github.com/AdoptOpenJDK/openjdk-support/issues
#
Call to MiniDumpWriteDump() failed (Error 0x80070006)

C:\Program Files\AdoptOpenJDK>
```

**Comment by jerboaa** [comment]
@StephanWald this seems to be a regression in 11.0.6 you are seeing. Fixed in 11.0.7+4 and better:
https://bugs.openjdk.java.net/browse/JDK-8238596
https://bugs.openjdk.java.net/browse/JDK-8238246

Work-around should be `-XX:UseAVX=0`

**Comment by Decat-SimonA** [comment]
Hello,
this happened in production, I'm not sure how to reproduce:

```
#  SIGSEGV (0xb) at pc=0x00007fc4403f21da, pid=16, tid=482
#
# Java VM: OpenJDK 64-Bit Server VM (12.0.2+10, mixed mode, sharing, tiered, compressed oops, serial gc, linux-amd64)
# J 24069 c2 sun.security.provider.NativePRNG$RandomIO.implNextBytes([B)V java.base@12.0.2 (162 bytes) @ 0x00007fc4403f21da [0x00007fc4403f1f00+0x00000000000002da]
Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
```

**Comment by karianna** [comment]
@Decat-SimonA the 12.x series is no longer supported - I'd move to 14.0.1

**Comment by Pengpengwanga** [comment]
Hello, I am using OpenJDK8 and I am also getting the similar crash
```shell
[CodeBlob (0x00007f1001045b50)]
Framesize: 84
Runtime Stub (0x00007f1001045b50): ic_miss_stub
Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (sharedRuntime.cpp:837), pid=8, tid=0x00007f0fe16e3700
#  fatal error: exception happened outside interpreter, nmethods and vtable stubs at pc 0x00007f1001045c06
#
# JRE version: OpenJDK Runtime Environment (8.0) (build 1.8.0-internal-_2020_04_23_16_32-b00)
# Java VM: OpenJDK 64-Bit Server VM (25.71-b00 mixed mode linux-amd64 compressed oops)
# Core dump written. Default location: //core or core.8
#
# An error report file with more information is saved as:
# //hs_err_pid8.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp

```

**Comment by karianna** [comment]
@Pengpengwanga - Can you open up a new issue with the full details, please.

**Comment by vmdarji85** [comment]
Hi All, I am getting similar error.

**Comment by aahlenst** [comment]
@vmdarji85 Please open a new issue in https://github.com/AdoptOpenJDK/openjdk-support and provide code that reproduces the problem. Otherwise we cannot help.
